engine/host_to_maddr.go

```diff
@@ -0,0 +1,43 @@
+package engine
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+
+	"github.com/multiformats/go-multiaddr"
+)
+
+func parseAddrType(host string) string {
+	addr := net.ParseIP(host)
+	if addr == nil {
+		re, _ := regexp.Compile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
+		if !re.MatchString(host) {
+			return ""
+		}
+		return "dns4"
+	}
+	if addr.To4() != nil {
+		return "ip4"
+	}
+	if addr.To16() != nil {
+		return "ip6"
+	}
+	return ""
+}
+
+func hostToMultiaddr(hostport string) (multiaddr.Multiaddr, error) {
+	host, port, err := net.SplitHostPort(hostport)
+	if err != nil {
+		port = "80"
+		host = hostport
+	}
```

Comment:
aren't there some forms of hostport that should actually error here?

Reply:
Errors should all be caught by `parseAddrType` and `multiaddr.NewMultiaddr`.  I added a unit test that checks failures.